### PR TITLE
(maint) Add better support for Cisco platforms

### PR DESCRIPTION
**Pull request by LuvCurves**
Need support for Cisco Virtrual Routing & Forwarding (VRF) as well as to use 'beaker' as the ssh username for Cisco NXOS

### CODE REVIEW

**Comment by LuvCurves** [comment]
@waynr Can you take a quick look?

**Comment by waynr** [comment]
:+1: This looks good to me. @LuvCurves would you mind updating the Changelog to indicate this as a bug fix in the "Unreleased" section?

@joshcooper @justinstoller, care to review and merge this once the changelog is updated? I'll trigger a patch-level release of the gem sometime afterward.

**Comment by joshcooper** [comment]
:+1:

File: lib/beaker-hostgenerator/data/vmpooler.rb [hunk 1]
```diff
@@ -38,10 +38,16 @@ module Vmpooler
         'cisconxos5-64' => {
           'platform' => 'cisco-5-x86_64',
           'template' => 'cisco-nxos-9k-x86_64'
+          'vrf' => 'management'
+          'ssh' => {
+            'auth_methods' => 'publickey'
```
Comment (justinstoller) [inline lib/beaker-hostgenerator/data/vmpooler.rb]: comma

File: lib/beaker-hostgenerator/data/vmpooler.rb [hunk 2]
```diff
@@ -38,10 +38,16 @@ module Vmpooler
         'cisconxos5-64' => {
           'platform' => 'cisco-5-x86_64',
           'template' => 'cisco-nxos-9k-x86_64'
+          'vrf' => 'management'
```
Comment (justinstoller) [inline lib/beaker-hostgenerator/data/vmpooler.rb]: comma

File: lib/beaker-hostgenerator/data/vmpooler.rb [hunk 3]
```diff
@@ -38,10 +38,16 @@ module Vmpooler
         'cisconxos5-64' => {
           'platform' => 'cisco-5-x86_64',
           'template' => 'cisco-nxos-9k-x86_64'
```
Comment (justinstoller) [inline lib/beaker-hostgenerator/data/vmpooler.rb]: comma

File: lib/beaker-hostgenerator/data/vmpooler.rb [hunk 4]
```diff
@@ -38,10 +38,16 @@ module Vmpooler
...
+          'vrf' => 'management'
+          'ssh' => {
+            'auth_methods' => 'publickey'
+            'user' => 'beaker'
+          }
         },
         'ciscoexr7-64' => {
           'platform' => 'cisco-7-x86_64',
           'template' => 'cisco-exr-9k-x86_64'
```
Comment (justinstoller) [inline lib/beaker-hostgenerator/data/vmpooler.rb]: comma ;)

**Comment by LuvCurves** [comment]
@justinstoller Commas added. Thx

**Comment by joshcooper** [comment]
Can we try running hiera acceptance against this PR? It should be possible doing:

```
$ export BEAKER_HOSTGENERATOR_VERSION=file:///Users/josh/work/beaker-hostgenerator
$ cd work/hiera/acceptance
$ bundle install --path .bundle
$ bundle exec rake ci:test:aio SHA=5e27c223c410a4d41de2716315e8682156278a80 SUITE_VERSION=1.3.2.203.g5e27c22 TEST_TARGET=cisconxos5-64a
```

At least to verify we can install puppet-agent and get through the presuite.

**Comment by LuvCurves** [comment]
@joshcooper @waynr Making sure the VM from the VMpooler is in the correct shape now.  We need some changes to Beaker too that I'm assisting @kevpl with at the moment.

**Comment by LuvCurves** [comment]
@joshcooper To test this correctly, we need a change to Beaker that will preface all commands being executed on the switch with 'sudo ip netns exec management '.  This is required to get the Cisco to use any networking (aka contacting the Puppet Master)

**Comment by joshcooper** [comment]
Beaker does something similar for executing puppet on windows, e.g. `cmd.exe /c puppet agent ...`

**Comment by LuvCurves** [comment]
Yes, Kevin is making sure it uses the same type of command prefacing as windows/cygwin

**Comment by LuvCurves** [comment]
This has been tested against Hiera, Facter and Puppet acceptance tests.  Ready for merge

**Pull request merged**